tests/test_additional_responses_router.py

```diff
@@ -33,6 +39,18 @@ async def c():
     return "c"
 
 
+@router.get(
+    "/d",
+    responses={
+        "400": {"description": "Error with str"},
+        "5xx": {"model": ResponseModel},
```

Comment:
this is lowercase, is there a conversion hidden somewhere? Your edit to `is_body_allowed_for_status_code` does not `upper()` it